src/components/Application/ApplicationStart.tsx

```diff
@@ -30,8 +37,27 @@ const ApplicationStart: React.FC<ApplicationStartProps> = (props) => {
     evaluate(startMessage || '', evaluatorParams).then((result: any) =>
       setStartMessageEvaluated(result)
     )
+
+    const isApplicationCompleted = Object.values(sectionsProgress).every(
+      ({ progress }) => progress?.completed
+    )
+    if (!isApplicationCompleted) findFirstImcompleteSection()
+
+    setIsApplicationCompleted(isApplicationCompleted)
   }, [startMessage, currentUser])
 
+  const findFirstImcompleteSection = () => {
+    const firstIncompleteSection = Object.entries(sectionsProgress)
```

Comment:
Also -- same comment as in the other file: if this is actually linking to a section AND page, perhaps call it `firstIncompleteLocation`?